src/sbf.cpp

```diff
@@ -0,0 +1,575 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file sbf.cpp
+ *
+ * Septentrio protocol as defined in PPSDK SBF Reference Guide 4.1.8
+ *
+ * @author Matej Frančeškin <Matej.Franceskin@gmail.com>
+ *
+*/
+
+#include "sbf.h"
+#include <string.h>
+#include <math.h>
+
+#define SBF_CONFIG_TIMEOUT    500      // ms, timeout for waiting ACK
+#define SBF_PACKET_TIMEOUT    2        // ms, if now data during this delay assume that full update received
+#define DISABLE_MSG_INTERVAL  1000000  // us, try to disable message with this interval
+
+/**** Trace macros, disable for production builds */
+#define SBF_TRACE_PARSER(...)   {/*GPS_INFO(__VA_ARGS__);*/}    /* decoding progress in parse_char() */
+#define SBF_TRACE_RXMSG(...)    {/*GPS_INFO(__VA_ARGS__);*/}    /* Rx msgs in payload_rx_done() */
+
+/**** Warning macros, disable to save memory */
+#define SBF_WARN(...)        {GPS_WARN(__VA_ARGS__);}
+#define SBF_DEBUG(...)       {/*GPS_WARN(__VA_ARGS__);*/}
+
+GPSDriverSBF::GPSDriverSBF(GPSCallbackPtr callback, void *callback_user,
+			   struct vehicle_gps_position_s *gps_position,
+			   struct satellite_info_s *satellite_info,
+			   uint8_t dynamic_model)
+	: GPSBaseStationSupport(callback, callback_user)
+	, _gps_position(gps_position)
+	, _satellite_info(satellite_info)
+	, _dynamic_model(dynamic_model)
+{
+	decodeInit();
+}
+
+GPSDriverSBF::~GPSDriverSBF()
+{
+}
+
+int
+GPSDriverSBF::configure(unsigned &baudrate, OutputMode output_mode)
+{
+	_configured = false;
+
+	// Check if we're already configured
+	setBaudrate(SBF_TX_CFG_PRT_BAUDRATE);
+
+	if (receive(SBF_CONFIG_TIMEOUT) > 0) {
+		return 0;
+	}
+
+	unsigned baud_i;
+
+	_output_mode = output_mode;
+	// try different baudrates
+	const unsigned baudrates[] = { 115200 };
+
+	for (baud_i = 0; baud_i < sizeof(baudrates) / sizeof(baudrates[0]); baud_i++) {
```

Comment:
This loop can be dropped since you're using a fixed baud.

Reply:
Yes, I dropped it. If we need it in future it is in git history.